File: scribbleshare-frontend/src/scripts/protocol/client/messages/ClientMessageCanvasMove.js
import ClientMessageType from "../ClientMessageType.js";
import ClientMessage from "../ClientMessage.js";

export default class ClientMessageCanvasMove extends ClientMessage {
    constructor(canvasMovesMap) {
        super(ClientMessageType.CANVAS_MOVE);
        this.canvasMovesMap = canvasMovesMap;
    }

    serialize(writer) {
        console.log('ceeck');
        super.serialize(writer);
        writer.writeUint8(this.canvasMovesMap.size);
        this.canvasMovesMap.forEach((canvasMoves, id) => {
            writer.writeInt16(id);
            writer.writeUint8(canvasMoves.length);
            canvasMoves.forEach((canvasMove) => {
                canvasMove.serialize(writer);
            });
        });
    }
}

File: scribbleshare-room/src/main/java/net/stzups/scribbleshare/room/server/websocket/protocol/client/messages/ClientMessageCanvasMove.java
package net.stzups.scribbleshare.room.server.websocket.protocol.client.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.scribbleshare.data.objects.canvas.CanvasMove;
import net.stzups.scribbleshare.room.server.websocket.protocol.client.ClientMessage;
import net.stzups.scribbleshare.room.server.websocket.protocol.client.ClientMessageType;

import java.util.HashMap;
import java.util.Map;

public class ClientMessageCanvasMove extends ClientMessage {

    private final Map<Short, CanvasMove[]> canvasMovesMap = new HashMap<>();

    public ClientMessageCanvasMove(ByteBuf byteBuf) {
        super(ClientMessageType.CANVAS_MOVE);
        int length = byteBuf.readUnsignedByte();
        for (int i = 0; i < length; i++) {
            short id = byteBuf.readShort();
            CanvasMove[] canvasObjectWrappers = new CanvasMove[byteBuf.readUnsignedByte()];
            for (int j = 0; j < canvasObjectWrappers.length; j++) {
                canvasObjectWrappers[j] = new CanvasMove(byteBuf);
            }
            canvasMovesMap.put(id, canvasObjectWrappers);
        }
    }

    public Map<Short, CanvasMove[]> getCanvasMovesMap() {
        return canvasMovesMap;
    }
}
